content/docs/rfcs/43/README.md

```diff
@@ -0,0 +1,347 @@
+in order to frequently and deterministically change the `messageNametag` of messages exchanged during the pairing and transfer phase - 
+ideally, at each message exchanged. 
+
+Given the proposed construction,
+the `ctsInbound` and `ctsOutbound` secrets can be used to iteratively generate the `messageNametag` field of Waku payloads 
+for inbound and outbound messages, respectively. 
+
+The derivation of `messageNametag` should be deterministic only for communicating devices 
+and independent from message content, 
+otherwise lost messages will prevent computing the next message nametag. 
+A possible approach consists in computing the `n`-th `messageNametag` as `H( ctsInbound || n)`, 
+where `n` is serialized as `uint64`.
+
+In this way, sender's and recipient's devices 
+can keep updated a buffer of `messageNametag` to sieve 
+while listening to messages sent over `/{application-name}/{application-version}/wakunoise/1/sessions-{shard-id}/` (i.e., the next 50 not yet seen). 
+They will then be able to further identify if one or more messages were eventually lost 
+or not-yet-delivered during the communication.
+This approach brings also the advantage that 
+communicating devices can efficiently identify encrypted messages addressed to them.
+
+We note that since the `ChaChaPoly` cipher used to encrypt messages supports *additional data*, 
+an encrypted payload can be further authenticated by passing the `messageNametag` as additional data to the encryption/decryption routine. 
+In this way, an attacker would be unable to craft an authenticated Waku message 
+even in case the currently used symmetric encryption key is compromised, 
+unless `ctsInbound`, `ctsOutbound` or the `messageNametag` buffer lists were compromised too.
+
+# Security/Privacy Considerations
+
+### Assumptions
+
+- The attacker is active, i.e. can interact with both devices `A` and `B` by sending messages over `contentTopic`.
+
+- The attacker has access to the QR code, that is knows the ephemeral key `eB`, the commitment `H(sB||r)` and the `contentTopic` exposed by the device `B`.
+
+- Devices `A` and `B` are considered trusted (otherwise the attacker will simply exfiltrate the relevant information from the attacked device).
+
+- As common for Noise, we assume that ephemeral keys cannot be compromised, while static keys might be later compromised. However, we enforce in the pairing phase extra security mechanisms (i.e. use of commitments for static keys) that will prevent some attacks possible when ephemeral keys are weak or get compromised.
+ 
+### Rationale
+
+- The device `B` exposes a commitment to its static key `sB` because:
+    - if the private key of `eB` is weak or gets compromised, an attacker can impersonate `B` by sending in message `c.` to device `A` his own static key and successfully complete the pairing phase. Note that being able to compromise `eB` is not contemplated by our security assumptions.
```

Comment:
AFAI understood, `eB` is transferred by QR code scan, right? then basically it does not get routed via Waku network. so even if it is weak, the attacker won't be able to see it, right? (I know it is part of the assumptions, but I am interested in the practical scenario and whether it is possible to get access to `eB`)

Reply:
Thanks for the clarification.
The way I see it is that (please let me know if I am missing something) even if `eB` is weak, the attacker can affect neither confidentiality nor the anonymity of the communication: 
- if an attacker finishes up part `c.` (assuming `eB` is compromised) yet it will get stuck in part `d.` where `sAsB` is needed (I mean DH of `sAsB`). Lacking the knowledge of `sB` (I mean DL of `sB`) means the attacker would not end up with a valid cipher state like Alice, hence I think he would not be able to compromise data privacy. So, there will be no data confidentially breach.
- As for anonymity in case the DL of `eB` is known to the attacker,  at stage `d.`, the attacker cannot get access to `sA` because that is encrypted under a key derived from `eAeB` and `eAsB`, so for decrypting `sA`, the attacker lacks the knowledge of `sB` (i.e., DL of `sB`). So, it does not seem to affect the anonymity either.

Reply:
> At this point `A` and `B` will observe the same `authcode` 
In the attack described above, I suspect that this happens i.e., that `At this point `A` and `B` will observe the same `authcode` ` given that the final authcode that is derived by `B` has a component of `eC` whereas  `A`'s auth code does not use that input. 

My point in general is to have a clear adversarial model and only discuss design choices related to that adversarial model. This makes it easier to reason about the security of the protocol. Nevertheless, if there are additional protections in the protocol to defeat a slightly stronger adversarial scenario, they can be mentioned separately, but with a _clear separation_ from the _original_ adversarial model.